tf2/include/tf2/buffer_core.h

```diff
@@ -317,19 +326,24 @@ class BufferCore
   
   /** \brief The pointers to potential frames that the tree can be made of.
    * The frames will be dynamically allocated at run time when set the first time. */
-  typedef std::vector<TimeCacheInterfacePtr> V_TimeCacheInterface;
-  V_TimeCacheInterface frames_;
-  
+  typedef std::map<CompactFrameID, TimeCacheInterfacePtr> M_TimeCacheInterface;
+  M_TimeCacheInterface frames_;
+
   /** \brief A mutex to protect testing and allocating new frames on the above vector. */
   mutable boost::mutex frame_mutex_;
 
   /** \brief A map from string frame ids to CompactFrameID */
   typedef boost::unordered_map<std::string, CompactFrameID> M_StringToCompactFrameID;
   M_StringToCompactFrameID frameIDs_;
   /** \brief A map from CompactFrameID frame_id_numbers to string for debugging and output */
-  std::vector<std::string> frameIDs_reverse;
+  typedef std::map<CompactFrameID, std::string> M_CompactFrameIDToString;
+  M_CompactFrameIDToString frameIDs_reverse;
+  /** \brief A frame counter used to increment the id of the newly created
+   * frames, even after deletion of some.
+  */
+  unsigned int frame_counter_;
```

Comment:
Can you add a comment to explain that frame_counter_ != frameIDs_.size() because it keeps increasing even if you clear a frame?